src/include/sof/audio/audio_stream.h

```diff
@@ -1040,6 +1040,14 @@ static inline void audio_stream_fmt_conversion(enum ipc4_bit_depth depth,
 	*frame_fmt = (depth >> 3) - 2;
 	*valid_fmt = (valid >> 3) - 2;
 
+	if (depth == 8) {
+		*frame_fmt = SOF_IPC_FRAME_U8;
+	}
+
+	if (valid == 8) {
+		*valid_fmt = SOF_IPC_FRAME_U8;
+	}
+
```

Comment:
Best to have the KCONFIG around this to save space.

Reply:
line 1043 - 1050 should ahead of line 1040, otherwise, "-2" operation will overflow.

Reply:
Ack @btian1 . If we can ensure both valid_fmt/frame_fmt are set (like in 24_3LE case), this is less of an issue as the invalid value gets overridden (and then we match the existing code layout for 24_3LE).

Reply:
There are many possible values of depth and valid (0..7, 9..15) for which this function returns a negative value and moving if upper will not correct it. The depth and valid variables is a enum and apparently the author assumed that there would be no illegal values. In the end, returning an invalid value won't break anything. It just won't find any conversion function.